portfolio/src/main/java/com/google/sps/servlets/ChartDataServlet.java

```diff
@@ -0,0 +1,40 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chart-data")
+public class ChartDataServlet extends HttpServlet {
+    private HashMap<String, Integer> chartVotes = new HashMap<>();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Gson gson = new Gson();
+        String json = gson.toJson(chartVotes);
+
+        response.setContentType("application/json");
+        response.getWriter().println(json);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String uglyChoice = request.getParameter("ugly-choice");
+        System.out.println(uglyChoice);
+        int currentVotes;
+
+        if (chartVotes.containsKey(uglyChoice)) {
+            currentVotes = chartVotes.get(uglyChoice);
+        } else {
+            currentVotes = 0;
+        }
+
+        chartVotes.put(uglyChoice, currentVotes + 1);
```

Comment:
why +1?

Reply:
Hmm i'm not following why you need to tally an extra vote on instead of showing the number of votes from he hashmap. Could you add a comment in the code to clarify?

Reply:
sure! there might be a better way to do this, but basically whenever the user votes for a certain category, this updates the map to now be +1 more than what it already was (before they voted).

Reply:
ah ok i misunderstood, i thought this was for displaying results. thanks for clarifying!